vendor/openfx/CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright Contributors to the OpenColorIO Project.
+
+set(SOURCES
+    OCIOColorSpace.cpp
+    OCIOMain.cpp
+    OCIOProcessor.cpp
+    OCIOUtils.cpp
+)
+set(OFXS_SOURCES
+    ${openfx_SUPPORT_DIR}/Library/ofxsCore.cpp
+    ${openfx_SUPPORT_DIR}/Library/ofxsImageEffect.cpp
+    ${openfx_SUPPORT_DIR}/Library/ofxsInteract.cpp
+    ${openfx_SUPPORT_DIR}/Library/ofxsLog.cpp
+    ${openfx_SUPPORT_DIR}/Library/ofxsMultiThread.cpp
+    ${openfx_SUPPORT_DIR}/Library/ofxsParams.cpp
+    ${openfx_SUPPORT_DIR}/Library/ofxsProperty.cpp
+    ${openfx_SUPPORT_DIR}/Library/ofxsPropertyValidation.cpp
+)
+set(PLIST
+    ${CMAKE_SOURCE_DIR}/share/openfx/Info.plist
+)
+set(RESOURCES
+    ${CMAKE_SOURCE_DIR}/share/openfx/resources/OpenColorIO.OCIOColorSpace.png
+    ${CMAKE_SOURCE_DIR}/share/openfx/resources/OpenColorIO.OCIOColorSpace.svg
+)
+
+add_library(ofxplugin MODULE ${SOURCES} ${OFXS_SOURCES})
+
+if(NOT BUILD_SHARED_LIBS)
+    target_compile_definitions(ofxplugin
+        PRIVATE
+            OpenColorIO_SKIP_IMPORTS
+    )
+endif()
+
+# Needed to define OpenFX Support Library visibility macros on Linux
+if(UNIX AND NOT APPLE)
+    target_compile_definitions(ofxplugin 
+        PRIVATE
+            linux
+    )
+endif()
+
+# These source files may be a future byproduct of OCIO_INSTALL_EXT_PACKAGES. 
+# Skip existence validation unless an explicit install location is provided.
+if(NOT DEFINED openfx_ROOT)
+    set_source_files_properties(${OFXS_SOURCES} PROPERTIES 
+        GENERATED TRUE)
+endif()
+
+# Disable known compiler warnings from OpenFX Support Library
+if(MSVC)
+    set(PLATFORM_COMPILE_FLAGS "${PLATFORM_COMPILE_FLAGS} /wd4996 /wd4101")
+else()
+    set(PLATFORM_COMPILE_FLAGS "${PLATFORM_COMPILE_FLAGS} -Wno-deprecated \
+                                                          -Wno-deprecated-declarations \
+                                                          -Wno-switch-enum \
+                                                          -Wno-dynamic-exception-spec \
+                                                          -Wno-catch-value \
+                                                          -Wno-unknown-warning-option")
+endif()
+
+# OpenFX Support Library source doesn't currently support C++17
+set(OFX_CXX_STANDARD ${CMAKE_CXX_STANDARD})
+if(${CMAKE_CXX_STANDARD} GREATER_EQUAL 17)
+    set(OFX_CXX_STANDARD 11)
+endif()
```

Comment:
That could not be here. For example the OCIO library compiles in C++17 but openfx compiles in C++11. That could be problematic even if we were careful in the OCIO public API. This constraints must be moved in the global CMakeLists.txt as a check when openfx is ON.

Reply:
Would it be worth maintaining a patch to have the OFX Support library C++17 compliant ?

Reply:
I was thinking something similar. I have a hunch that most of the changes would be pretty boilerplate (i.e. replace `auto_ptr` with `unique_ptr`). I'll do some investigation into this. The Support Library source appears to have the same BSD 3-Clause license as OCIO, but ideally updates would be contributed back to openfx in the long term.

Reply:
See note below, but I have removed the CMake commands in question here